plot.go

```diff
@@ -439,49 +439,71 @@ func (p *Plot) NominalY(names ...string) {
 	p.Y.Tick.Marker = ConstantTicks(ticks)
 }
 
-// Save saves the plot to an image file.  The file format is determined
-// by the extension.
+// WriterTo returns an io.WriterTo that will write the plot as
+// the specified image format.
 //
-// Supported extensions are:
+// Supported formats are:
 //
-//  .eps, .jpg, .jpeg, .pdf, .png, .svg, and .tiff.
-func (p *Plot) Save(w, h vg.Length, file string) (err error) {
+//  eps, jpg|jpeg, pdf, png, svg, and tif|tiff.
+func (p *Plot) WriterTo(w, h vg.Length, format string) (io.WriterTo, error) {
 	var c interface {
-		vg.Canvas
-		Size() (w, h vg.Length)
+		vg.CanvasSizer
 		io.WriterTo
 	}
-	switch ext := strings.ToLower(filepath.Ext(file)); ext {
-
-	case ".eps":
-		c = vgeps.NewTitle(w, h, file)
+	switch format {
+	case "eps":
+		c = vgeps.New(w, h)
 
-	case ".jpg", ".jpeg":
+	case "jpg", "jpeg":
 		c = vgimg.JpegCanvas{Canvas: vgimg.New(w, h)}
 
-	case ".pdf":
+	case "pdf":
 		c = vgpdf.New(w, h)
 
-	case ".png":
+	case "png":
 		c = vgimg.PngCanvas{Canvas: vgimg.New(w, h)}
 
-	case ".svg":
+	case "svg":
 		c = vgsvg.New(w, h)
 
-	case ".tiff":
+	case "tif", "tiff":
 		c = vgimg.TiffCanvas{Canvas: vgimg.New(w, h)}
 
 	default:
-		return fmt.Errorf("Unsupported file extension: %s", ext)
+		return nil, fmt.Errorf("unsupported format: %q", format)
 	}
 	p.Draw(draw.New(c))
 
+	return c, nil
+}
+
+// Save saves the plot to an image file.  The file format is determined
+// by the extension.
+//
+// Supported extensions are:
+//
+//  .eps, .jpg, .jpeg, .pdf, .png, .svg, .tif and .tiff.
+func (p *Plot) Save(w, h vg.Length, file string) (err error) {
 	f, err := os.Create(file)
 	if err != nil {
 		return err
 	}
-	if _, err = c.WriteTo(f); err != nil {
+	defer func() {
+		e := f.Close()
```

Comment:
I like it, but I can understand why you may not. I won't bug about it in future PRs.